package/upgrade/lib.sh

```diff
@@ -571,6 +571,57 @@ EOF
   wait_for_addon_upgrade_deployment $name $namespace $enabled $curstatus
 }
 
+upgrade_harvester_upgradelog_with_patch_loggingref()
+{
+  local namespace=$UPGRADE_NAMESPACE
+  local upgradelogname=$(kubectl get upgrades.harvesterhci.io $HARVESTER_UPGRADE_NAME -n $UPGRADE_NAMESPACE -ojsonpath="{.status.upgradeLog}")
+  local loggingref="harvester-upgradelog"
+
+  if [[ -z "$upgradelogname" ]]; then
+    echo "upgradelog is not found from upgrade $HARVESTER_UPGRADE_NAME, nothing to do"
+    return 0
+  fi
```

Comment:
This script es executed with `-e` aka `-o errexit`, meaning that if any command fails (return code non-zero) the script will exit.

In case the `kubectl` command above returns non-zero, this will completely abort the script and the `if` block will never be reached.
This can happen for a variety of reasons, e.g. the API server may respond with HTTP code 503 and a Retry-After header if it's under heavy load.

You may want to turn off error exit for this passage using `set -e`, but don't forget to turn it back on later with `set +e` ;-)

Reply:
It is slightly different, if you use `local` to set a var via a command output, if the kubectl/yq... returns any error, the error code is escaped and the `local`ed var is empty.

You may try this.

Reply:
You're right. I missed that. :facepalm: